__tests__/unittest/templates.test.js

```diff
@@ -0,0 +1,197 @@
+const cds = require('@sap/cds');
+const templates = require('../../lib/templates');
+
+describe('templates', () => {
+    let linkedModel;
+
+    beforeEach(() => {
+        linkedModel = cds.linked(`
+            namespace customer.testNamespace123;
+            entity Books {
+                key ID: UUID;
+                title: String;
+            }
+        `);
+    }); 
+    
+    describe('fCreateEntityTypeTemplate', () => {
+        it('should return default value', () => {
+            expect(templates.fCreateEntityTypeTemplate(linkedModel)).toEqual({
+                // TODO: fix undefined. Root cause: get the value from ${entity['@ODM.entityName']} without setting default value
+                ordId: 'sap.odm:entityType:undefined:v1'
+            });
+        });
+    });
+
+    describe('checkEntityFunctionAction', () => {
+        it('should return entity', () => {
+            const testResult = [
+                {
+                    type: 'entity',
+                    name: 'customer.testNamespace123.Books' ,
+                    entityType: 'customer.testNamespace123.Books',
+                    entitySet: 'customer.testNamespace123.Books',
+                    entityTypeMapping: 'undefined:entityType:customer.testNamespace123.Books:v1',
+                    entitySetMapping: 'undefined:entitySet:customer.testNamespace123.Books:v1',
+                }
+            ]
+            expect(templates.checkEntityFunctionAction(linkedModel, global)).toEqual(testResult);
+        });
+
+        it('should return actions', () => {
+        });
+
+        it('should return function', () => {
+        });
+
+    })
+
+    describe('fReplaceSpecialCharacters', () => {
+        it('should return replaced dot', () => {
+            const testString = 'customer.testNamespace.123';
+            expect(templates.fReplaceSpecialCharacters(testString)).toEqual('customer.testNamespace123');
+        });
+
+        it('should return replaced dash', () => {
+            const testString = 'customer.testNamespace-123';
+            expect(templates.fReplaceSpecialCharacters(testString)).toEqual('customer.testNamespace123');
+        });
+
+        // TODO: is that correct?
```

Comment:
Just a proposal: Let's remove the TODO comments out and move them in a GitHub issue instead

Reply:
Yes, will do